### ISSUE TITLE
Adding exclusions setting

### ISSUE THREAD

**Issue by maxfilov** [form-com/diff-coverage-gradle#69]
It would be nice to have a specific setting which specifies the files which should be thrown out of coverage check. This helps in cases when `@Data` from lombok library is used. One could argue that those methods should still be tested, but it would still be a nice feature.

I suggest adding something like
```groovy
    diffCoverageReport {
        diffSource.git.compareWith 'refs/remotes/origin/develop'

        violationRules.failIfCoverageLessThan 0.8

        reports {
            html = true
        }
        
        exclusions = "**/MyClass.java", "**/model/*.java" # // <-- This is the setting I'm talking about
    }
```
@SurpSG could you please provide your opinion on this matter? I think I can implement the necessary code myself, but I want to get your confirmation first.

UPD: It looks like this option is present in maven version of this plugin, it means that this should not be a problem to implement it for gradle as well.

**Comment by SurpSG** [comment]
Hi @maxfilov 

Yep, agree. Exclusion of classes/packages is nice to have

**Comment by k3marek** [comment]
Hi @SurpSG, the DiffCoverage is a great tool but I think that for full success it lacks the excluding feature. 
I found that the diff coverage checks classes that were excluded from the main jacoco report from which the execs are taken.
What's your ETA for this feature?

**Comment by SurpSG** [comment]
Hu @k3marek 

If you required the feature ASAP then you could apply the next workaround:
```groovy
diffCoverageReport {

    // how to exclude:
    def allClassFiles = subprojects.jacocoTestReport.classDirectories // 1. Collect all class files from jacocoTestReport tasks
    def classFilesAsFileTree = files(subprojects.jacocoTestReport.classDirectories).getAsFileTree() // 2. Convert FileCollection to the FileTree
    classesDirs = classFilesAsFileTree.matching { // 3. pass the filtered collection to diff coverage plugin
        // Define excludes
        exclude '**/com/form/coverage/gradle/**/*', '**/other/**/*'
    }
    // -----------------------------------
   
   // the same but inlined:
    classesDirs = files(subprojects.jacocoTestReport.classDirectories).getAsFileTree().matching {
        exclude '**/com/form/coverage/gradle/**/*', '**/other/**/*'
    }

}
```
Diff-Coverage analyzes only classes that were provided to the plugin. All other classes are considered as ignored.

About the feature implementation ETA. It could take up to 2 months. I guess the feature will be included into the next planned major release `1.0.0`

**Comment by SurpSG** [comment]
Will be available in release 1.0.0
